src/providers/MeetingProvider/types.ts

```diff
@@ -44,4 +44,6 @@ export interface ManagerConfig {
   logLevel: LogLevel;
   postLogConfig?: PostLogConfig;
   simulcastEnabled?: boolean;
+  consoleLogger?: ConsoleLogger;
```

Comment:
This can be any logger just not console logger. Can we just use the interface `Logger` here?

Reply:
It was `Logger` before. But Venky said I should handle the case where user pass a `postLogConfig` and `consoleLogger` at the same time. Check the code [here](https://github.com/aws/amazon-chime-sdk-component-library-react/blob/8a19883caa8fcbf638e3e0fa7c10cbf684c099cf/src/providers/MeetingProvider/MeetingManager.ts#L242-L254)

Reply:
@devalevenkatesh , could you please check here? Am I following your suggestion correctly?

Reply:
Hmm, so this should be `logger?: Logger`. I meant in the `createLogger` function I suggested to override the existing `ConsoleLogger` creation when `logger` is passed in the `meetingConfig`.

Reply:
Oh, how about changing this
```js
// Current Changes
createLogger(configuration: MeetingSessionConfiguration) {
  let consoleLogger: ConsoleLogger;

  if (this.consoleLogger) {
    consoleLogger = this.consoleLogger;
  } else {
    consoleLogger = new ConsoleLogger('SDK', this.logLevel);
  }

  let logger: ConsoleLogger | MultiLogger = consoleLogger;
  this.meetingSession = new DefaultMeetingSession(
    configuration,
    logger,
    deviceController,
    this.eventReporter
  );
```
to
```js
// After
createLogger(configuration: MeetingSessionConfiguration) {
  // Change to `Logger`, as well as the one in `MeetingConfig`
  let consoleLogger: Logger;

  if (this.consoleLogger) {
    consoleLogger = this.consoleLogger;
  } else {
    consoleLogger = new ConsoleLogger('SDK', this.logLevel);
  }

  let logger: ConsoleLogger | MultiLogger = consoleLogger;
  this.meetingSession = new DefaultMeetingSession(
    configuration,
    logger,
    deviceController,
    this.eventReporter
  );
```

Reply:
So when a user passes a `logger` and `postLogConfig`, you would handle both the logger creation in the `createLogger` using `MultiLogger`. It would be something like below:
```js
createLogger(configuration: MeetingSessionConfiguration) {
    let logger: Logger;

    if (this.postLoggerConfig) {
      logger = new MultiLogger(
        this.logger ? this.logger : new ConsoleLogger('SDK', this.logLevel),
        new MeetingSessionPOSTLogger(
          this.postLoggerConfig.name,
          configuration,
          this.postLoggerConfig.batchSize,
          this.postLoggerConfig.intervalMs,
          this.postLoggerConfig.url,
          this.postLoggerConfig.logLevel
        )
      );
    }
    return logger;
  }
```

Reply:
I remember there is a type conflict, let me double check and I will update if possible